app/src/main/java/com/lagradost/cloudstream3/ui/result/ResultViewModel2.kt

```diff
@@ -1684,6 +1695,39 @@ class ResultViewModel2 : ViewModel() {
                 }
             }
 
+            ACTION_FCAST -> {
+                val devices = FcastManager.currentDevices.toList()
+                postPopup(
+                    txt(R.string.player_settings_select_cast_device),
+                    devices.map { txt(it.name) }) { index ->
+                    if (index == null) return@postPopup
+                    val device = devices.getOrNull(index)
+
+                    acquireSingleLink(
+                        click.data,
+                        LoadType.Chromecast,
```

Comment:
This should be its own param as it will use the noreferer param in the extraction due to `isCasting = LoadType.Chromecast == type`